client/src/js/TinyMCE_ssmedia.js

```diff
@@ -88,34 +84,60 @@
             });
         },
 
+        cloudinaryInstance() {
+            return new Cloudinary({
+                cloud: {
+                    cloudName: CLOUDINARY_CONFIG.cloud_name,
+                }
+            });
+        },
+
         /**
          * Inserts the data into the editor
          *
          * @param {*} response
          */
-        insertHandler(response) {
+        insertImageHandler(response) {
             const asset = response.assets[0];
+console.log('insertImageHandler', asset);
             if (asset.resource_type !== 'image') {
                 throw `Resource type of [${asset.resource_type}] is not supported`;
             }
 
-            let secureUrl;
-            if (asset.derived && asset.derived.length > 0) {
-                secureUrl = asset.derived[0].secure_url;
-            } else {
-                secureUrl = asset.secure_url;
+            const image = this.cloudinaryInstance().image(asset.public_id);
+            const defaultTransformations = this.editor.getParam('default_transformations');
+
+            if (defaultTransformations) {
+                const transformationString = transformationStringFromObject(defaultTransformations);
+                image.addTransformation(transformationString);
+            }
+
+            // Copied same logic from `API::extractDescription()`
+            const defaultAltText = asset.context && asset.context.custom && asset.context.custom.alt ? asset.context.custom.alt : '';
```

Comment:
JS supports optional chaining, so this could be simplified to asset.context?.custom?.alt ?? '';